mail_multicompany_force_company/models/ir_mail_server.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, models
+from odoo.osv import expression
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        domain = []
+        if not self._context.get("show_all"):
+            domain = [
+                "|",
+                ("company_id", "=", self.env.company.id),
+                ("company_id", "=", False),
+            ]
```

Comment:
It seems to me a bit risky to do it this way: other parts of Odoo, or modules, might depend on the search returning mail servers for all companies, and you'd change that behavior globally.

Wouldn't it be better to override [find_mail_server](https://github.com/OCA/OCB/blob/361d7db2ea75b1d01623ea0d40d4f371fd78ac99/odoo/addons/base/models/ir_mail_server.py#L652) and inject a context key `dont_return_all` there, that you catch here and if it's there, you limit the results?